src/libLookup/Lookup.cpp

```diff
@@ -1148,21 +1149,19 @@ void Lookup::RetrieveTxBlocks(vector<TxBlock>& txBlocks, uint64_t& lowBlockNum,
   lock_guard<mutex> g(m_mediator.m_node->m_mutexFinalBlock);
 
   if (lowBlockNum == 0) {
-    // give all the blocks till now in blockchain
     lowBlockNum = 1;
-  } else {
-    uint64_t lowestLimitNum = (m_mediator.m_dsBlockChain.GetBlockCount() >
-                               INCRDB_DSNUMS_WITH_STATEDELTAS)
-                                  ? (m_mediator.m_dsBlockChain.GetLastBlock()
-                                         .GetHeader()
-                                         .GetBlockNum() -
-                                     INCRDB_DSNUMS_WITH_STATEDELTAS) *
-                                        NUM_FINAL_BLOCK_PER_POW
-                                  : 0;
-    if (lowBlockNum <= lowestLimitNum) {
-      // Limit the number of txn blocks upto last N ds epochs
-      lowBlockNum = lowestLimitNum;
-    }
+  }
+
+  uint64_t lowestLimitNum =
+      (m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum() - 1) *
+      NUM_FINAL_BLOCK_PER_POW;
```

Comment:
It's better to use `GetEpochNum()` in case of any change of NUM_FINAL_BLOCK_PER_ROW in the future